Cli/Helper/ConfigurationHelper.php

```diff
@@ -84,20 +88,24 @@ private function getValue(InputInterface $input, $varName)
 
     private function askValue(InputInterface $input, OutputInterface $output, $varname, $default = null)
     {
-        if ($default) {
-            $question = sprintf('What is your %s? [%s] ', $varname, $default);
-        } else {
-            $question = sprintf('What is your %s? ', $varname);
-        }
-
-        $validator = function ($v) {
+        $validator = function ($v) use ($varname) {
             if (!$v) {
-                throw new \InvalidArgumentException('Your must answer this question.');
+                throw new \InvalidArgumentException(sprintf('Your must provide a %s!', $varname));
             }
 
             return $v;
         };
 
+        if (!$input->isInteractive()) {
+            return call_user_func($validator, $default);
```

Comment:
This syntax does not work for all callables, which is why it is not a good idea to use it all the time. But here, it is indeed safe as the validator callable is known.

For reminder, here are the supported rules:

- on 5.3, it supports only closures
- on 5.4+, it supports all callables *except* ``array($className, $staticMethodName)``